routes/web.php

```diff
@@ -29,6 +29,13 @@
     Route::get('about', 'AboutController@index');
     Route::get('authlog', 'UserController@authlog');
 
+    // Maps
+    Route::group(['prefix' => 'maps', 'namespace' => 'Maps'], function () {
+        Route::get('devicedependency', 'DeviceDependencyController@dependencyMap');
+        Route::get('devicedependency/group={group_id}', 'DeviceDependencyController@dependencyMap');
```

Comment:
Lets either use query strings for this, or maybe better, create a new function in your controller to handle this.  
One for normal dependency, one for group, and one function that renders the map.
```php
Route::get('devicedependency/group/{group}', 'DeviceDependencyController@groupDependency');
```

```php
function groupDependency(Request $request, DeviceGroup $group) {
  // get devices...
  return self::renderMap($devices);
}
```

Reply:
group and highlight_node are now classical get Parameter